### DISCUSSION TITLE
建议 Repository Update 中增加一个是否强制更新 JsonMap 字段的参数

### DISCUSSION THREAD

**Discussion by woodcoal** [dotnetcore/FreeSql#1151] [Ideas]
参考问题：#280 
Repository Update 下 JsonMap 字段无法自动更新，建议增加一个参数是否强制更新 JsonMap 字段，毕竟 JsonMap 字段是否更新，使用的人最清楚。

如： int Update(TEntity entity, bool withJsonMap);
withJsonMap 设置为 true 时强制更新 JsonMap 字段

**Comment by 2881099** [comment]
这个问题确实有点坑，只要是引用类型都不行。

粗爆解决方法，如果指定必更新可以 Attach 后重新 new 一次，或者 尝试 Clone 一次：

entity.JsonMapProperty = new

**Reply by woodcoal** [reply]
Clone 的想法我也试过，但是这样势必会增加损耗，而且克隆还存在一个深度的问题。
个人倾向于增加开关，强制保存一次 JsonMap 字段

**Reply by 2881099** [reply]
按照现有对比机制，啰嗦的写法如下：

```c#
var tmpJsonMap = entity.JsonMap;
entity.JsonMap = null;
repo.Attach(entity);
entity.JsonMap = tmpJsonMap;
repo.Update(entity);
```

**Reply by 2881099** [reply]
可以考虑增加一个方法，直接修改 State
